### PR TITLE
Hyundai: car port for Sonata 2024 (CAN FD)

### PR DESCRIPTION
**Pull request by Kirito3481**
**Checklist**
- [x] added entry to CarInfo in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [ ] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: `78ad5150de133637|2023-11-29--21-08-18`
- [x] route with stock system: `78ad5150de133637|2023-11-29--20-26-11`
- [x] Harness Type: [Hyundai A](https://github.com/commaai/openpilot/wiki/Hyundai-Kia-Genesis#harness-guide)
- [x] Architecture: CAN-FD

### CODE REVIEW

**Comment by sshane** [comment]
Can you plug in your comma power and start the car? Interested in getting a more complete set of FW versions. fwdRadar says `RDR`, but there's no ADAS ECU, like other newer HKGs.

**Comment by Kirito3481** [comment]
> Can you plug in your comma power and start the car? Interested in getting a more complete set of FW versions. fwdRadar says `RDR`, but there's no ADAS ECU, like other newer HKGs.

I plugged in the comma power and booted, but the ADAS ECU was not detected.

Route: '78ad5150de133637|2023-12-01--17-13-41'

**Comment by sshane** [comment]
Does this port just work? No steering faults, persistent dash faults, etc?

File: selfdrive/car/torque_data/substitute.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ HYUNDAI ELANTRA HYBRID 2021: HYUNDAI SONATA 2020
 HYUNDAI TUCSON 2019: HYUNDAI SANTA FE 2019
 HYUNDAI TUCSON 4TH GEN: HYUNDAI TUCSON HYBRID 4TH GEN
 HYUNDAI SANTA FE 2022: HYUNDAI SANTA FE HYBRID 2022
+HYUNDAI SONATA 2024: HYUNDAI SONATA 2020
```
Comment (sshane) [inline selfdrive/car/torque_data/substitute.yaml:37]: use override, [2.5, 2.5, 0.1]

**Comment by sshane** [comment]
- [ ] Can you get a test route of about 10 minutes, using openpilot engaged?

**Comment by Kirito3481** [comment]
> * [ ]  Can you get a test route of about 10 minutes, using openpilot engaged?
> 
>     * [ ]  Is this HDA1 or HDA2, or do they not brand it like that where you are?

Route using openpilot: `78ad5150de133637|2023-12-02--01-53-09`
Route using sunnypilot: `78ad5150de133637|2023-12-01--21-59-04`

In the price catalog it is marked HDA1.

**Comment by sshane** [comment]
What happens if you try to turn on alpha longitudinal?

File: selfdrive/car/hyundai/values.py [hunk 2]
```diff
@@ -2114,7 +2124,8 @@ def match_fw_to_car_fuzzy(live_fw_versions) -> Set[str]:
 CANFD_CAR = {CAR.KIA_EV6, CAR.IONIQ_5, CAR.IONIQ_6, CAR.TUCSON_4TH_GEN, CAR.TUCSON_HYBRID_4TH_GEN, CAR.KIA_SPORTAGE_HYBRID_5TH_GEN,
              CAR.SANTA_CRUZ_1ST_GEN, CAR.KIA_SPORTAGE_5TH_GEN, CAR.GENESIS_GV70_1ST_GEN, CAR.KIA_SORENTO_PHEV_4TH_GEN,
              CAR.GENESIS_GV60_EV_1ST_GEN, CAR.KIA_SORENTO_4TH_GEN, CAR.KIA_NIRO_HEV_2ND_GEN, CAR.KIA_NIRO_EV_2ND_GEN,
-             CAR.GENESIS_GV80, CAR.KIA_CARNIVAL_4TH_GEN, CAR.KIA_SORENTO_HEV_4TH_GEN, CAR.KONA_EV_2ND_GEN, CAR.KIA_K8_HEV_1ST_GEN}
+             CAR.GENESIS_GV80, CAR.KIA_CARNIVAL_4TH_GEN, CAR.KIA_SORENTO_HEV_4TH_GEN, CAR.KONA_EV_2ND_GEN,
+             CAR.KIA_K8_HEV_1ST_GEN, CAR.SONATA_2024}
 
 # The radar does SCC on these cars when HDA I, rather than the camera
 CANFD_RADAR_SCC_CAR = {CAR.GENESIS_GV70_1ST_GEN, CAR.KIA_SORENTO_PHEV_4TH_GEN, CAR.KIA_SORENTO_4TH_GEN, CAR.GENESIS_GV80,
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:2131]: Let's test alpha long with Sonata 2024 both in this set, and not in this set. If it doesn't work, add to `CANFD_UNSUPPORTED_LONGITUDINAL_CAR`

Comment (sshane) [reply selfdrive/car/hyundai/values.py:2131]: Any updates?

Comment (Kirito3481) [reply selfdrive/car/hyundai/values.py:2131]: > Any updates?

I can't take a car now. I think it will be possible from Thursday.

Comment (sshane) [reply selfdrive/car/hyundai/values.py:2143]: I updated your branch so alpha longitudinal should be available. Can you try that and report back?

**Comment by sshane** [comment]
Any problems with the car with this port? I see torqued learned respectable values (2.2~ m/s^2).

Other new Hyundais are having a few issues relating to bad tuning/oversteering, faults on the dash, messages telling them to take control, etc. It doesn't sound like you are experiencing any of these issues, right?

**Comment by elibb98** [comment]
@sshane I have a C3X and a Sonata 2024 SEL. Anything I can do to help port the car?

**Comment by sshane** [comment]
Can you outline the experience with this branch and your car?

- Can it steer comfortable (any tuning issues)?
- Any faults or unexpected dash error lights?
- Any other issues?

**Comment by elibb98** [comment]
I definitely can. Do I install this branch Kirito3481:sonata-2024-port? Or is there a way to install just this pr?

**Pull request closed without merging**